src/test/ui/suggestions/unnecessary_dot_for_floating_point_literal.stderr

```diff
@@ -0,0 +1,52 @@
+error[E0308]: mismatched types
+  --> $DIR/unnecessary_dot_for_floating_point_literal.rs:2:18
+   |
+LL |     let _: f64 = 0..10;
+   |            ---   ^^^^^ expected `f64`, found struct `std::ops::Range`
+   |            |
+   |            expected due to this
+   |
+   = note: expected type `f64`
+            found struct `std::ops::Range<{integer}>`
+help: remove the unnecessary `.` operator to to use a floating point literal
+   |
+LL -     let _: f64 = 0..10;
+LL +     let _: f64 = 0.10;
+   |
+
+error[E0308]: mismatched types
+  --> $DIR/unnecessary_dot_for_floating_point_literal.rs:3:18
+   |
+LL |     let _: f64 = 0..;
+   |            ---   ^^^ expected `f64`, found struct `RangeFrom`
+   |            |
+   |            expected due to this
+   |
+   = note: expected type `f64`
+            found struct `RangeFrom<{integer}>`
+
+error[E0308]: mismatched types
+  --> $DIR/unnecessary_dot_for_floating_point_literal.rs:4:18
+   |
+LL |     let _: f64 = ..10;
```

Comment:
The problem here is that it is possible that one wanted to write `.10`, which is not valid rust, but still used in some places of the world to represent `0.10`. So I think we should suggest `0.12` when someone wrote `..12` (or present `12.` as a possibility as well).

Also, `0.` should be fine and that should get the suggestion as well.

Reply:
Today, when people write `.1`, we output:

```
error: float literals must have an integer part
 --> src/main.rs:2:18
  |
2 |     let x: f64 = .1;
  |                  ^^ help: must have an integer part: `0.1`
```

I agree that ideally the two cases highlighted should provide a suggestion.